gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/AuditServiceImpl.java

```diff
@@ -49,7 +50,9 @@ public class AuditServiceImpl implements AuditService {
     @Override
     public Single<Page<Audit>> search(ReferenceType referenceType, String referenceId, AuditReportableCriteria criteria, int page, int size) {
         try {
-            return getReporter(referenceType, referenceId).search(referenceType, referenceId, criteria, page, size);
+            return Maybe.fromOptional(getReporter(referenceType, referenceId))
+                    .flatMapSingle(r -> r.search(referenceType, referenceId, criteria, page, size))
+                    .switchIfEmpty(Single.just(new Page<>()));
```

Comment:
**suggestion**: to avoid useless rx subscription and publisher convertion, why not doing something like this ?

`getReporter(x,y).map(r -> r.search(referenceType, referenceId, criteria, page, size)).orElse(Single.just(new Page))`